honeycombio/data_source_columns.go

```diff
@@ -0,0 +1,90 @@
+package honeycombio
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	honeycombio "github.com/honeycombio/terraform-provider-honeycombio/client"
+	"github.com/honeycombio/terraform-provider-honeycombio/honeycombio/internal/hashcode"
+)
+
+func dataSourceHoneycombioColumns() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceHoneycombioColumnsRead,
+
+		Schema: map[string]*schema.Schema{
+			"dataset": {
+				Type:     schema.TypeString,
+				Optional: false,
+				Required: true,
+			},
+			"starts_with": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"names": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"types": {
```

Comment:
I don't love the idea of a separate list of types where we hope that the two lists are sharing the same index.

What I'd like to see instead is the pattern emerging in the community where datasources that are plural return a list of identifiers (names in this case) which can then be passed to a singular data source to get a fully enriched object with all of the properties set. In this case, that would mean the addition of a `honeycombio_column` data source which takes a name.

This would allow something like this to give full objects with all the fields set:

```hcl
data "honeycombio_columns" "all" {
  dataset = var.dataset
}

data "honeycombio_column" "all" {
  for_each = data.honeycombio_columns.names
  
  dataset = var.dataset
}
```

And you can now use this to check `data.honeycombio_column.all["mycolumn"].type`.

This does mean a bit more code (with the second datasource addition), but I think the flexibility is worth it and it plays nicely with the emerging idiom in the community.

Reply:
okay phew, I was really hating that output 🙂 
will implement a `honeycombio_column` datasource as well